app/job/rest.py

```diff
@@ -53,6 +55,15 @@ def get_job_by_service_and_job_id(service_id, job_id):
     return jsonify(data=data)
 
 
+@job.route('/<job_id>/cancel', methods=['POST'])
+def cancel_job(service_id, job_id):
+    job = dao_get_future_scheduled_job_by_id_and_service_id(job_id, service_id)
+    job.job_status = JOB_STATUS_CANCELLED
+    dao_update_job(job)
+
+    return get_job_by_service_and_job_id(service_id, job_id)
```

Comment:
You don't need to do another query to get the job object, it has been updated by the dao_update_job(job).
Also it's probably better to return json version of the object:
return jsonify(data=job_schema.dump(job).data)